.github/workflows/test.yml

```diff
@@ -5,22 +5,32 @@ concurrency:
   cancel-in-progress: true
 
 on:
-  pull_request:
+  pull_request_target:
     branches: [main]
+    types: [labeled]
   workflow_dispatch:
 
 jobs:
   changelog:
+    if: contains(github.event.pull_request.labels.*.name, 'safe to test')
```

Comment:
Could you put this test in one job and `needs: label_check` (or whatever) the rest?

Reply:
yes, great idea

Reply:
follow on Pr coming